app/controllers/demo_mode/sessions_controller.rb

```diff
@@ -26,6 +26,7 @@ def new
     def create
       @session = Session.new(create_params)
       @session.save_and_generate_account_later!(**options_params.to_unsafe_h.deep_symbolize_keys)
+      @session.reload if ActiveJob::Base.queue_adapter_name == 'test'
```

Comment:
@smudge temporarily gating this with the `:test` adapter, then a follow-up to update the global to use `:test` and the actual inline one to use `:inline`